fix: close SQL result sets immediately in JobsToSchedule loop

analytically opened the pull request:
The JobsToSchedule method was using defer Close() inside the job iteration loop, causing all result sets to remain open until function completion. With many jobs to schedule, this exhausts database connections and memory.

Changed to close resourceRows, resourceTypeRows, and prototypeRows immediately after processing each job instead of deferring closure until function return. Added rows.Err() checks after each iteration loop for defensive error handling.

taylorsilva reviewed (changes requested):
By not calling `Close()` with `defer` this presents a new risk: resource leaks if the function returns before we can call `.Close()`.

See https://github.com/golang/go/issues/46863

Good news is, we can call `.Close()` multiple times without any side-effects! Please add back the `defer Close()` lines, then this will be good to merge.

analytically committed:
fix: close SQL result sets immediately in JobsToSchedule loop

The JobsToSchedule method was using defer Close() inside the job
iteration loop, causing all result sets to remain open until function
completion. With many jobs to schedule, this exhausts database
connections and memory.

Changed to close resourceRows, resourceTypeRows, and prototypeRows
immediately after processing each job instead of deferring closure
until function return. Added rows.Err() checks after each iteration
loop for defensive error handling.

Signed-off-by: Mathias Bogaert <mathias.bogaert@gmail.com>

taylorsilva reviewed (approved):
LGTM

Nit: the `.Close()` inside the `rows.Err()` if statements is redundant because the deferred `Close()` will be called when we hit the `return nil, err`.
